vital/tests/test_geo_point.cxx

```diff
@@ -54,139 +53,78 @@ static auto constexpr crs_utm_18n = kwiver::vital::SRID::UTM_WGS84_north + 18;
 int
 main(int argc, char* argv[])
 {
-  CHECK_ARGS(1);
   kwiver::vital::plugin_manager::instance().load_all_plugins();
 
-  testname_t const testname = argv[1];
-
-  RUN_TEST(testname);
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
 }
 
 // ----------------------------------------------------------------------------
-IMPLEMENT_TEST(default_constructor)
+TEST(geo_point, default_constructor)
 {
   kwiver::vital::geo_point p;
-
-  if ( ! p.is_empty() )
-  {
-    TEST_ERROR("The default point is not empty");
-  }
+  EXPECT_TRUE( p.is_empty() );
 }
 
 // ----------------------------------------------------------------------------
-IMPLEMENT_TEST(constructor_point)
+TEST(geo_point, constructor_point)
 {
   kwiver::vital::geo_point p{ loc1, crs_ll };
-
-  if ( p.is_empty() )
-  {
-    TEST_ERROR("The constructed point is empty");
-  }
+  EXPECT_FALSE( p.is_empty() );
 }
 
 // ----------------------------------------------------------------------------
-IMPLEMENT_TEST(assignment)
+TEST(geo_point, assignment)
 {
   kwiver::vital::geo_point p;
   kwiver::vital::geo_point const p1{ loc1, crs_ll };
   kwiver::vital::geo_point const p2;
 
-  if ( ! p.is_empty() )
-  {
-    TEST_ERROR("The default point is not empty");
-  }
+  // Paranoia-check initial state
+  EXPECT_TRUE( p.is_empty() );
```

Comment:
I assume the GTest output provides enough context (file and line number) in the output to distinguish which test is failing.  For example, line 96 below has the identical test, but in a different context.  In converting to GTest you are discarding the error message strings that provided more context to error.  Are we okay with that?  This is more of a group opinion poll than a particular suggestion.  I guess if I can track a failure back to a particular line in file that is sufficient.  But I might immediate know what is wrong if I get more context in the failure message.

Reply:
Yes; that's one of the good things about Google Test; assertions contain the location of the failure, and generally both the text arguments as well as evaluated values of the predicate. For example, after monkeying with the preceding code so that this assertion fails, I get:

    path/to/kwiver/vital/tests/test_geo_point.cxx:84: Failure
    Value of: p.is_empty()
      Actual: false
    Expected: true


My philosophy is generally to assume that if an assertion fails, you are at least going to look at that assertion in the source code, and that redundantly providing information that is readily apparent from that context is, well, redundant :smile:. Mostly because this significantly cuts down on the time it takes to *write* unit tests. That's not to say I'd be strongly opposed to other philosophies, however.

(Code-wise, it is trivial to add a more detailed explanation: `EXPECT_<PRED>(...) << "details"`. However, what do I write for "details"? "Default constructed point should be empty"? "Default constructed point is empty"? "Default constructed point is *not* empty"? Do I say "point", or "geo_point"? "vital::geo_point"? Do I capitalize it? End with a period? Already with `TEST_ERROR` we aren't fully consistent in answering such questions. Taking the approach that we only add details when they aren't obvious from looking at the assertion in its source code context allows these questions to be ignored 95% of the time.)

Reply:
I guess it doesn't matter much as long as the test code is documented well enough to be clear what the test is trying to test.  I can imagine someone writing a long sequence of `EXPECT_*` statements without documenting what the test is doing.  The old framework forced your to write *something* for each test.  That said, commentary is not always needed for every test call.  A comment covering a block of tests, as you've done here, is probably sufficient.